syntax_and_semantics/offsetof.md

```diff
@@ -0,0 +1,20 @@
+# offsetof
+
+The `offsetof` instruction returns the byte offset of an instance variable in a struct or class type. It accepts the type as first argument and the instance variable name prefixed by an `@` as second argument:
```

Comment:
Let's stay consistent and refer to them as "expressions", just like in the other 3 expression docs.
```suggestion
The `offsetof` expression returns the byte offset of an instance variable in a struct or class type. It accepts the type as the first argument and the instance variable name prefixed by an `@` as the second argument:
```

Reply:
I don't think we should describe them as "expressions". `offsetof` itself is not an expression in Crystal. A call like `offsetof(Type, @foo)` is an expression.
I'm open to alternative suggestions.

Reply:
Keyword?

Reply:
I would go with "expression". `offsetof` nodes are transformed into number literals at compile time, which certainly are expressions. "Instruction" would be a bit misleading here since there is no actual instruction generated.

Reply:
Maybe ``An `offsetof` expression``, replacing `The` by `An` to show that it's not only `offsetof` but kinda an 'instance' of offsetof, like `offsetof(Foo, @bar)`

Reply:
@malte-v Yes, `offsetof` nodes are expressions, there's no debate. But this doesn't relate to a concrete expression as a call to `offsetof`, but the `offsetof` construct itself. It's similar to a method (or rather a macro method) but I don't think we have a specific name for this.

Reply:
I would go with "Keyword" or "Primitive" (this term is used on https://crystal-lang.org/reference/syntax_and_semantics/low_level_primitives.html).

Reply:
I think that 

> An `offset` expression ... 

is accurate enough. We are not saying that the sole keyword is the expression.

Other than that this is GTG.